lib/CFRPageActions.jsm

```diff
@@ -68,12 +84,46 @@ class PageAction {
     this._showPopupOnClick = this._showPopupOnClick.bind(this);
     this.dispatchUserAction = this.dispatchUserAction.bind(this);
 
-    this._l10n = new DOMLocalization(["browser/newtab/asrouter.ftl"]);
+    this._l10n = this._createDOML10n();
 
     // Saved timeout IDs for scheduled state changes, so they can be cancelled
     this.stateTransitionTimeoutIDs = [];
   }
 
+  /**
+   * Creates a new DOMLocalization instance with the Fluent file from Remote Settings.
+   * Note that it still uses the packaged Fluent file as a fallback.
+   */
+  _createDOML10n() {
+    async function* generateBundles(resourceIds) {
+      const appLocale = Services.locale.appLocaleAsBCP47;
+      const appLocales = Services.locale.appLocalesAsBCP47;
+      // The downloaded Fluent file is located in
+      // `/${OS.Constants.Path.localProfileDir}/settings/main/ms-language-packs/`
+      const l10nFluentDir = OS.Path.join(
+        OS.Constants.Path.localProfileDir,
+        "settings",
+        "main",
+        "ms-language-packs"
+      );
```

Comment:
This could be defined as a constant. 
Also, the downloader helper returns the file URI already. And since it's a no-op when the file exists, you can use it to obtain the URI

Reply:
Yep, I've noticed that downloader would return the file URI upon successful downloading. However, it's kinda cumbersome for us to pass or persist that URI so that it can be used later by this module. Also, given that the file URI is located in a well-known directory of RS, so I've chosen to assemble it over here.